### ISSUE TITLE
Number of classes

### ISSUE THREAD

**Issue by ronghui1992** [xqchen/Scalable-Construction-of-Compressed-Gene-sets-from-Single-cell-Data-Using-Active-Feature-Selection#2]
Dear author,

I am trying  your code on 2 class (cell type) problem and it fails, but it works if I get 3 classes. Is it possible to run on 2 class data?

**Comment by xqchen** [comment]
Yes, it should work for binary classification tasks. What are the errors when you try binary classification? BTW, I have the latest version of this algorithm in another repository called activeSVC. [https://github.com/xqchen/activeSVC](url)

**Comment by ronghui1992** [comment]
I am using the example from min_complexity.ipynb and changing the target to binary classification task
The error goes like this:
ValueError: This solver needs samples of at least 2 classes in the data, but the data contains only one class: 2
And it works if there are 3 classes in the target.

Also, I set markers to 2 but the results could changed, is that because of the random selection cells ?

**Comment by ronghui1992** [comment]
Or could I use seurat object on this?

**Comment by xqchen** [comment]
I have a notice for this error in the latest version of this algorithm in the repository link I just shared as: Sometimes it shows ValueError: This solver needs samples of at least 2 classes in the data, but the data contains only one class: 0. This is because of the settings of algorithm. Please set a larger num_samples and init_samples and then re-run this cell. Or set parameter class_weight='balanced' and balance=True and then re-run the cell.

**Comment by ronghui1992** [comment]
Thanks a lot, I will try this. 
I run the example code and get different results with the markers as 2. Should I do anything else to fix it?
